lib/browser/platform/ios.rb

```diff
@@ -3,10 +3,21 @@ module Browser
   class Platform
     class IOS < Base
       MATCHER = /(iPhone|iPad|iPod)/
-      VERSION_MATCHER = /OS ([\d.]+)/
+      VERSION_MATCHER = /OS ((?<major>\d)_(?<minor>\d)_?(?<patch>\d)?)/
 
       def version
-        ua[VERSION_MATCHER, 1] || "0"
+        matches = VERSION_MATCHER.match(ua)
```

Comment:
Can't you use something like the following?

```ruby
matches = VERSION_MATCHER.match(ua)
return "0" unless matches
matches.to_a[1..-1].join(".")
```

Also, why are you skipping MINOR when is zero?